### ISSUE TITLE
Push State: Navigation to the current page should noop

### ISSUE THREAD

**Issue by giuseppeg** [Automattic/wp-calypso#4008]
Hello there,
I found a small bug in Calypso's push state navigation which I believe being trivial to fix.

Basically when one clicks on a link that points to the same page a new state is added to the browser history and therefore one has to click the back button multiple times (fire a popstate event) to navigate back to the previous view.
#### Steps to reproduce
1. Starting at URL: https://wordpress.com
2. Click on any link e.g. the profile link
3. Wait until the new view is loaded e.g. https://wordpress.com/me
4. Click again the profile link a few times
5. Now try to go back i.e. click the browser's back button
#### What I expected

I expect to get to the previous page after one back button click.
Basically:

``` javascript
const currentUrl = window.location.pathname + window.location.search;
if (currentTarget.getAttribute('href') === currentUrl) {
  // noop
  return;
}
```
#### What happened instead

The page remains the same, and I get to the previous page only after `n` clicks where `n` is the number of clicks from `4.`.
#### Browser / OS version

Any / OS X Yosemite Version 10.10.5
#### Screenshot / Video

https://youtu.be/ag4JVNP6lOs

**Comment by lancewillett** [comment]
CC @mtias — worth fixing? Nice UX improvement on the back button.

**Comment by hoverduck** [comment]
#bug-scrub

Confirmed this is still an issue -

![wp-calypso-4008](https://cloud.githubusercontent.com/assets/7233112/17902372/1dccdd0c-6935-11e6-9fa5-f83aa70f0299.gif)

**Comment by hoverduck** [comment]
#bug-scrub

Still an issue as described.  @mtias - Is it as easy to fix as the OP suggests?  Assuming it is I'm going to add the `Good First Change` label so hopefully somebody can pick it up.

**Comment by iandunn** [comment]
This appears to be an upstream issue in page.js, and possibly the intended behavior.

https://github.com/visionmedia/page.js/issues/417
https://github.com/visionmedia/page.js/issues/366

417 has a workaround, but the complexity it would add to the codebase seems worse than the problem it solves.

I think we should close this until page.js implements a simple way to achieve it. Please feel free to reopen it if there's a simpler way and I just missed it.
